packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js

```diff
@@ -0,0 +1,256 @@
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_loading_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  addConfig() {
+    this.config.interface = '';
+    this.config.leaseDuration = '';
+    this.config.offset = '';
+    this.config.poolSize = '';
+    this.config.priority = '';
+    this.showAddConfig = true;
+    this.displayAddConfigButton = false;
+  }
+
+  createConfig() {
+    this.errorMessage = '';
+    this.successMessage = '';
+    if (
+      !this.config.leaseDuration &&
+      !this.config.offset &&
+      !this.config.poolSize &&
+      !this.config.priority
+    ) {
+      return this.$q.resolve(null);
+    }
+
+    const params = {
+      configuration: {
+        leaseDuration: this.config.leaseDuration,
+        offset: this.config.offset,
+        poolSize: this.config.poolSize,
+      },
+      priority: this.config.priority,
+    };
+
+    if (this.config.interface) {
+      params.configuration.interface = this.config.interface;
+    }
+
+    return this.$http
+      .post(`/overTheBox/${this.serviceName}/configuration/dhcp/config`, params)
+      .then((response) => {
+        this.successMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_dhcp_config_create_succeed',
+        );
+
+        // Reload config table
+        this.loadConfig();
+        this.showAddConfig = false;
+        this.displayAddConfigButton = true;
+        return response;
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_create_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  cancelCreateConfig() {
+    this.showAddConfig = false;
+    this.displayAddConfigButton = true;
+  }
+
+  removeConfig(row) {
+    this.errorMessage = '';
+    this.successMessage = '';
+    return this.$http
+      .delete(
+        `/overTheBox/${this.serviceName}/configuration/dhcp/config/${row.id}`,
+      )
+      .then((response) => {
+        // Display success message
+        this.successMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_dhcp_config_remove_succeed',
+        );
+
+        // Reload config table
+        this.loadConfig();
+        return response;
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_remove_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  loadStaticLeases() {
+    this.staticLeases = [];
+    return this.$http
+      .get(`/overTheBox/${this.serviceName}/configuration/dhcp/staticLease`)
+      .then((response) => {
+        if (response.data) {
+          this.staticLeases = response.data;
+        }
```

Comment:
`response.data` should be at least empty, and if no `response.data`, we are in error case no?
```suggestion
          this.staticLeases = response.data;
```